exchange/exchange_test.go

```diff
@@ -2540,13 +2548,164 @@ func TestCategoryMappingTwoBiddersOneBidEachNoCategorySamePrice(t *testing.T) {
 		} else {
 			assert.Nil(t, seatBidApn1.bids, "Appnexus_1 seat bid should not have any bids back")
 			assert.Len(t, seatBidApn2.bids, 1, "Appnexus_2 seat bid should have only one back")
+		}
+	}
+}
+
+func TestCategoryMappingTwoBiddersManyBidsEachNoCategorySamePrice(t *testing.T) {
+	// This test covers a very rare de-duplication case where bid needs to be removed from already processed bidder
+	// This happens when current processing bidder has a bid that has same de-duplication key as a bid from already processed bidder
+	// and already processed bid was selected to be removed
+
+	//In this test case bids bid_idApn1_1 and bid_idApn1_2 will be removed due to hardcoded "fakeRandomDeduplicateBidBooleanGenerator{true}"
```

Comment:
Thank you. If there a reason we need `bid_idApn1_1` and `bid_idApn1_2` to model this situation? Would just `bid_idApn1_1` also be a valid test? Or do we need to the 2 placements for the bidder to already be processed? which if so, could you make a bit clearer in the description?

Reply:
We need to have more than one bid in each bidder to test how we remove element from array. If there is only one element in bidder, then we remove the entire bidder. 
I added explicit comment for this.